source/extensions/filters/http/gzip/gzip_filter.cc

```diff
@@ -123,16 +128,20 @@ Http::FilterHeadersStatus GzipFilter::encodeHeaders(Http::HeaderMap& headers, bo
     headers.insertContentEncoding().value(Http::Headers::get().ContentEncodingValues.Gzip);
     compressor_.init(config_->compressionLevel(), config_->compressionStrategy(),
                      config_->windowBits(), config_->memoryLevel());
-  } else {
+
+  } else if (!skip_compression_) {
     skip_compression_ = true;
+    stats_.not_compressed_.inc();
   }
-
   return Http::FilterHeadersStatus::Continue;
 }
 
 Http::FilterDataStatus GzipFilter::encodeData(Buffer::Instance& data, bool end_stream) {
   if (!skip_compression_) {
+    stats_.total_uncompressed_bytes_.add(data.length());
     compressor_.compress(data, end_stream ? Compressor::State::Finish : Compressor::State::Flush);
+    stats_.total_compressed_bytes_.add(data.length());
+    stats_.compressed_.inc();
```

Comment:
I think the compressed stat would be more useful if it were per-request, not per data frame similar to not_compressed?

Reply:
Yea, Jose and I thought that the delta per data frame would be good enough, but happy to change it to per request. if you have a good example of how to do that, that would be awesome.

Reply:
I would probably just invert the logic such that when you know that you are not "not compressing" you would increment the compressing stat. Not sure of the best way of doing that since I haven't looked at the filter code in detail, but I'm sure you can sort something out.

Reply:
sounds good.

Reply:
ah nvm, i see what you mean. Yup. will fix to move this into the encodeHeaders